anta/result_manager/models.py

```diff
@@ -111,16 +118,83 @@ def _set_status(self, status: AntaTestStatus, message: str | None = None) -> Non
             Status of the test.
         message
             Optional message.
-
         """
         self.result = status
+        if (self._parent.result == AntaTestStatus.UNSET and status == AntaTestStatus.SUCCESS) or status in [AntaTestStatus.FAILURE, AntaTestStatus.ERROR]:
+            self._parent.result = status
         if message is not None:
             self.messages.append(message)
+            self._parent.messages.append(message)
+
+
+class TestResult(BaseTestResult):
+    """Describe the result of a test from a single device.
+
+    Attributes
+    ----------
+    name : str
+        Name of the device on which the test was run.
+    test : str
+        Name of the AntaTest subclass.
+    categories : list[str]
+        List of categories the TestResult belongs to. Defaults to the AntaTest subclass categories.
+    description : str
+        Description of the TestResult. Defaults to the AntaTest subclass description.
+    inputs:  BaseModel
+        Inputs of the AntaTest instance.
+    custom_field : str | None
+        Custom field to store a string for flexibility in integrating with ANTA.
+    result : AntaTestStatus
+        Result of the test.
+    messages : list[str]
+        Messages reported by the test.
+    atomic_results: list[AtomicTestResult]
+        A list of AtomicTestResult instances which can be used to store atomic results during the test execution.
+        It can then be leveraged in the report to render atomic results over the test global TestResult.
+    """
+
+    name: str
+    test: str
+    categories: list[str]
+    description: str
+    inputs: SerializeAsAny[InstanceOf[BaseModel]] | None = None  # A TestResult inputs can be None in case of inputs validation error
+    custom_field: str | None = None
+    result: AntaTestStatus = AntaTestStatus.UNSET
+    messages: list[str] = []
+    atomic_results: list[AtomicTestResult] = []
 
     def __str__(self) -> str:
         """Return a human readable string of this TestResult."""
         return f"Test '{self.test}' (on '{self.name}'): Result '{self.result}'\nMessages: {self.messages}"
```

Comment:
shoyld this change to indicate it has AtomicTestResults?

Reply:
Do you have an example where this could be needed?